File: assurance/fixtures/sdlc/python/tests/verify.py
from pathlib import Path

import os

import idna


root = Path(__file__).resolve().parents[1]
if idna.__version__ != "3.10":
    raise SystemExit(f"expected locked idna 3.10, got {idna.__version__}")
environment = Path(os.environ["VIRTUAL_ENV"]).resolve()
if not Path(idna.__file__).resolve().is_relative_to(environment):
    raise SystemExit("idna was not imported from the locked task environment")
expected = "3.10:xn--fa-hia.de\n"
actual = (root / "generated/value.txt").read_text()
if actual != expected:
    raise SystemExit("prepared Python consumer output does not match its input")

File: assurance/fixtures/sdlc/pnpm/tests/verify.mjs
import { readFile } from "node:fs/promises";
import { readFileSync } from "node:fs";
import { createRequire } from "node:module";
import { dirname, join } from "node:path";
import kleur from "kleur";

const require = createRequire(import.meta.url);
if (JSON.parse(readFileSync(join(dirname(require.resolve("kleur")), "package.json"))).version !== "4.1.5") {
  throw new Error("expected locked kleur 4.1.5");
}
kleur.enabled = true;
if (kleur.bold("locked") !== "\u001b[1mlocked\u001b[22m") {
  throw new Error("locked kleur did not produce the expected ANSI output");
}
const expected = "\u001b[1mpnpm-only consumer payload\u001b[22m";
const actual = await readFile("generated/value.txt", "utf8");
if (actual !== expected) {
  throw new Error("prepared pnpm consumer output does not match its input");
}
